pkg/client/retrieve.go

```diff
@@ -49,25 +49,27 @@ func (c *SonobuoyClient) RetrieveResults(cfg *RetrieveConfig) io.Reader {
 	}, scheme.ParameterCodec)
 	executor, err := remotecommand.NewSPDYExecutor(c.RestConfig, "POST", req.URL())
 	if err != nil {
-		cfg.Errc <- fmt.Errorf("unable to get remote executor: %v", err)
-		return nil
+		return nil, err
 	}
 	reader, writer := io.Pipe()
-	// We use a goroutine here to allow this function to return a reader that lets the caller
-	// deal with what to do with this stream of data and keep that detail out of this function.
-	go func() {
+	go func(writer *io.PipeWriter) {
 		defer writer.Close()
-
 		err = executor.Stream(remotecommand.StreamOptions{
 			Stdout: writer,
-			Stderr: cfg.CmdErr,
+			Stderr: os.Stderr,
 			Tty:    false,
 		})
 		if err != nil {
-			cfg.Errc <- fmt.Errorf("error streaming: %v", err)
+			// Since this function returns an io.Reader to the consumer and does
```

Comment:
From a client perspective, how would the error manifest itself?  

Would the reader just error when the writer closes?